app/models/tag.rb

```diff
@@ -158,7 +158,7 @@ def can_be_archived?
   end
 
   def can_be_removed?
-    level_two? && draft?
+    raise NotImplementedError
```

Comment:
We shouldn't use this type of [Error](https://oleg0potapov.medium.com/ruby-notimplementederror-dont-use-it-dff1fd7228e5) - it'd be better to raise an Error with a message?

Reply:
@AgaDufrat The rest looks good to me so if you can flip this change in some point I can approve 👍

Reply:
oh, interesting. I first read about the template method pattern (and the use of `NotImplementedError` for it) in Sandi Metz’s “Practical Object-Oriented Design in Ruby” https://cek.io/blog/2014/05/06/poodr-ch-6
I'll change it here and in other places.

Reply:
I've also been using `NotImplementedError` but I don't want to override Kelv on it (we may have internal precedence).

Reply:
Response from one of our Lead Devs is that they avoid using `NotImplementedError` for this context.